onnxruntime/test/onnx/TestCase.cc

```diff
@@ -52,100 +52,111 @@
+  return ret;
 }
 
-template <>
-Status Convert(const google::protobuf::Map<int64_t, float>& data, MapInt64ToFloat** out) {
-  ConvertMap(data, out);
-  return Status::OK();
+template <typename key_type, typename value_type>
+OrtValue* PbMapToMLValue(const google::protobuf::Map<key_type, value_type>& map) {
+  OrtAllocatorInfo* info;
+  ORT_THROW_ON_ERROR(OrtCreateAllocatorInfo("Cpu", OrtDeviceAllocator, 0, OrtMemTypeDefault, &info));
+  std::unique_ptr<OrtAllocatorInfo, decltype(&OrtReleaseAllocatorInfo)> rel_info(info, OrtReleaseAllocatorInfo);
+  const size_t ele_count = map.size();
+  std::vector<int64_t> dims(1, ele_count);
+  std::vector<key_type> keys(ele_count);
+  std::vector<value_type> values(ele_count);
+  size_t i = 0;
+  for (auto& kvp : map) {
+    keys[i] = kvp.first;
+    values[i] = kvp.second;
+    ++i;
+  }
+  std::vector<OrtValue*> map_in(2);
+  map_in[0] = CreateTensorWithDataAsOrtValue(info, keys);
+  if (map_in[0] == nullptr) ORT_THROW("Create keys tensor failed");
+
+  map_in[1] = CreateTensorWithDataAsOrtValue(info, values);
+  if (map_in[1] == nullptr) ORT_THROW("Create values tensor failed");
+
+  // create map ort value
+  OrtValue* map_ort = nullptr;
+  ORT_THROW_ON_ERROR(OrtCreateValue(map_in.data(), 2, ONNX_TYPE_MAP, &map_ort));
+  for (auto value : map_in) OrtReleaseValue(value);
+  return map_ort;
 }
 
-template <>
-Status Convert(const google::protobuf::Map<int64_t, double>& data, MapInt64ToDouble** out) {
-  ConvertMap(data, out);
-  return Status::OK();
-}
-template <typename InputType, typename OutputType>
-Status RichTypeProtoToMLValue(const InputType& input, MLValue& value) {
-  OutputType* tensor = nullptr;
-  Status st = Convert(input, &tensor);
-  if (!st.IsOK()) return st;
-  value.Init(tensor,
-             DataTypeImpl::GetType<OutputType>(),
-             DataTypeImpl::GetType<OutputType>()->GetDeleteFunc());
-  return Status::OK();
+template <typename T>
+void VectorProtoToMLValue(const RepeatedPtrField<T>& input, ORT_VALUE_HOLDER& output) {
+  OrtAllocatorInfo* info;
+  ORT_THROW_ON_ERROR(OrtCreateAllocatorInfo("Cpu", OrtDeviceAllocator, 0, OrtMemTypeDefault, &info));
+  std::unique_ptr<OrtAllocatorInfo, decltype(&OrtReleaseAllocatorInfo)> rel_info(info, OrtReleaseAllocatorInfo);
+  std::vector<OrtValue*> in(input.size());
+  size_t j = 0;
+  for (const T& v : input) {
+    // create key tensor
+    const auto& map = v.v();
+    size_t ele_count = map.size();
+    using key_type = typename std::remove_reference<decltype(v.v())>::type::key_type;
+    using value_type = typename std::remove_reference<decltype(v.v())>::type::mapped_type;
+    std::vector<int64_t> dims(1, static_cast<int64_t>(ele_count));
+    std::vector<key_type> keys(ele_count);
+    std::vector<value_type> values(ele_count);
+    size_t i = 0;
+    for (auto& kvp : map) {
+      keys[i] = kvp.first;
+      values[i] = kvp.second;
+      ++i;
+    }
+    std::vector<OrtValue*> map_in(2);
```

Comment:
Better to store a vector of objects that will release OrtValue at the end, than doing it manually using "for (auto value : in) OrtReleaseValue(value);".